static/app/components/charts/baseChart.tsx

```diff
@@ -348,84 +355,90 @@ function BaseChartUnwrapped({
   isGroupedByDate = false,
   transformSinglePointToBar = false,
   transformSinglePointToLine = false,
-  onChartReady = () => {},
+  onChartReady = DEFAULT_CHART_READY,
   'data-test-id': dataTestId,
 }: BaseChartProps) {
   const theme = useTheme();
 
-  const hasSinglePoints = (series as LineSeriesOption[] | undefined)?.every(
-    s => Array.isArray(s.data) && s.data.length <= 1
-  );
-
   const resolveColors =
     colors !== undefined ? (Array.isArray(colors) ? colors : colors(theme)) : null;
+
   const color =
     resolveColors ||
     (series.length ? theme.charts.getColorPalette(series.length) : theme.charts.colors);
-  const previousPeriodColors =
-    previousPeriod && previousPeriod.length > 1 ? lightenHexToRgb(color) : undefined;
-
-  const transformedSeries =
-    (hasSinglePoints && transformSinglePointToBar
-      ? (series as LineSeriesOption[] | undefined)?.map(s => ({
-          ...s,
-          type: 'bar',
-          barWidth: 40,
-          barGap: 0,
-          itemStyle: {...(s.areaStyle ?? {})},
-        }))
-      : hasSinglePoints && transformSinglePointToLine
-      ? (series as LineSeriesOption[] | undefined)?.map(s => ({
-          ...s,
-          type: 'line',
-          itemStyle: {...(s.lineStyle ?? {})},
-          markLine:
-            s?.data?.[0]?.[1] !== undefined
-              ? MarkLine({
-                  silent: true,
-                  lineStyle: {
-                    type: 'solid',
-                    width: 1.5,
-                  },
-                  data: [{yAxis: s?.data?.[0]?.[1]}],
-                  label: {
-                    show: false,
-                  },
-                })
-              : undefined,
-        }))
-      : series) ?? [];
-
-  const transformedPreviousPeriod =
-    previousPeriod?.map((previous, seriesIndex) =>
-      LineSeries({
-        name: previous.seriesName,
-        data: previous.data.map(({name, value}) => [name, value]),
-        lineStyle: {
-          color: previousPeriodColors ? previousPeriodColors[seriesIndex] : theme.gray200,
-          type: 'dotted',
-        },
-        itemStyle: {
-          color: previousPeriodColors ? previousPeriodColors[seriesIndex] : theme.gray200,
-        },
-        stack: 'previous',
-        animation: false,
-      })
-    ) ?? [];
 
-  const resolvedSeries = !previousPeriod
-    ? [...transformedSeries, ...additionalSeries]
-    : [...transformedSeries, ...transformedPreviousPeriod, ...additionalSeries];
-
-  const defaultAxesProps = {theme};
+  const resolvedSeries = useMemo(() => {
+    const previousPeriodColors =
+      previousPeriod && previousPeriod.length > 1 ? lightenHexToRgb(color) : undefined;
```

Comment:
```suggestion
      previousPeriod?.length > 1 ? lightenHexToRgb(color) : undefined;
```

Reply:
I need to `previousPeriod?.length ?? 0` then since I'm comparing to int. No strong preferences here tbh